### DISCUSSION TITLE
Community.VisualStudio.SourceGenerators

### DISCUSSION THREAD

**Discussion by kaheimri** [VsixCommunity/Community.VisualStudio.Toolkit#380] [Q&A]
Hi,
first of all many thanks for this awesome project. We are looking into converting our VS extension to use the Vsix Community Toolkit.
So far it looks quite promising.  My question is about the VSIX Manifest and Command Table file [source generators](https://github.com/VsixCommunity/Community.VisualStudio.SourceGenerators). Our extensions shall support the 3 latest products, VS 2017, 2019 and VS 2022, but we did not get the generator working with VS 2017. Is VS 2017 a supported use case for the generator?

BR, Karsten

**Comment by madskristensen** [comment]
The generator isn't available for VS 2017, but that only matters if you are using VS 2017 to code on your extension. Is that what you do?

**Reply by kaheimri** [reply]
We do develop on VS 2022, though we have a setup (1) to build the extension via the version we are targeting. The reason for this is we have to build and deploy parts of Qt with that specific version (2), to be able to use our custom .dll that wraps QML debugging via a COM interface. 

(1) A .bat file setting up the environment, figuring out the VS version, fetching all nuget packages and run some code generation on T4 text templates...
(2) We build Qt as x86 for VS2017/2019 and x64 for VS2022. 

I guess you're hinting towards using e.g. VS2019 to target VS2017, right?

**Comment by reduckted** [comment]
The output from source generators can be saved to disk. 

https://learn.microsoft.com/en-us/dotnet/csharp/roslyn-sdk/source-generators-overview

> You can also set build properties to save the generated file and control where the generated files are stored. In the console application's project file, add the `<EmitCompilerGeneratedFiles>` element to a `<PropertyGroup>`.

If you can control the order of your build, you could build the 2019 or 2022 extension first, and then include the generated files in the build for the 2017 extension.
